components/dash-table/src/dash-table/dash/DataTable.js

```diff
@@ -110,15 +110,18 @@ export const defaultProps = {
 
 export const propTypes = {
     /**
-     * The row and column indices and IDs of the currently active cell.
-     * `row_id` is only returned if the data rows have an `id` key.
+     * The contents of the table.
+     * The keys of each item in data should match the column IDs.
+     * Each item can also have an 'id' key, whose value is its row ID. If there
+     * is a column with ID='id' this will display the row ID, otherwise it is
+     * just used to reference the row for selections, filtering, etc.
+     * Example:
+     * [
+     *      {'column-1': 4.5, 'column-2': 'montreal', 'column-3': 'canada'},
+     *      {'column-1': 8, 'column-2': 'boston', 'column-3': 'america'}
+     * ]
      */
-    active_cell: PropTypes.exact({
-        row: PropTypes.number,
-        column: PropTypes.number,
-        row_id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-        column_id: PropTypes.string
-    }),
+    data: PropTypes.arrayOf(PropTypes.object),
```

Comment:
Maybe we could be a bit more descriptive/specific with the type, like defining it as something like `PropTypes.objectOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number]))`.

Reply:
I think it's a thing to consider and good suggestion! But here are 2 things in my mind.
- Having object can give more flexibility to support formulas or reference to another cell like in Excel? Not even sure if this is a designed feature. :)
- I tried to not touch existing stuff that can potentially break anything. 😆 

@chriddyp What would you say?

Reply:
We don't support objects as values yet. So yeah string, number, bool would be fine as far as I know. And our tests should pick things up if not.